Doc/library/os.rst

```diff
@@ -4625,17 +4590,122 @@ written in Python, such as a mail server's external command delivery program.
-   as those provided to :func:`waitpid`.
+   process id, exit status indication, and resource usage information is
+   returned.  Refer to :func:`resource.getrusage` for details on resource usage
+   information.  The arguments to :func:`wait4` are the same as those provided
+   to :func:`waitpid`.
 
    :func:`waitstatus_to_exitcode` can be used to convert the exit status into an
    exitcode.
 
    .. availability:: Unix, not Emscripten, not WASI.
 
 
+.. data:: P_PID
+          P_PGID
+          P_ALL
+          P_PIDFD
+
+   These are the possible values for *idtype* in :func:`waitid`. They affect
+   how *id* is interpreted:
+
+   * :data:`P_PID` - wait for the child whose PID is *id*.
+   * :data:`P_PGID` - wait for any child whose progress group ID is *id*.
+   * :data:`P_ALL` - wait for any child; *id* is ignored.
+   * :data:`P_PIDFD` - wait for the child identified by the file descriptor
+     *id* (a process file descriptor created with :func:`pidfd_open`).
+
+   .. availability:: Unix, not Emscripten, not WASI.
+
+   .. note:: ``P_PIDFD`` is only available on Linux >= 5.4.
+
+   .. versionadded:: 3.3
+   .. versionadded:: 3.9
+      The ``P_PIDFD`` constant.
+
+
+.. data:: WCONTINUED
+
+   This option for :func:`waitpid`, :func:`wait3`, :func:`wait4`, and
+   :func:`waitid` causes child processes to also be reported if they have been
+   continued from a job control stop since their status was last reported.
+
+   .. availability:: Unix, not Emscripten, not WASI.
+
+
+.. data:: WEXITED
+
+   This option for :func:`waitid` causes child processes that have terminated to
+   be reported.
+
+   The other ``wait*`` function always report children that have terminated,
+   so this option is not available for them.
+
+   .. availability:: Unix, not Emscripten, not WASI.
+
+   .. versionadded:: 3.3
+
+
+.. data:: WSTOPPED
+
+   This option for :func:`waitid` causes child processes that have been stopped
+   by the delivery of a signal to be reported.
+
+   This option is not available for the other ``wait*`` functions.
+
+   .. availability:: Unix, not Emscripten, not WASI.
+
+   .. versionadded:: 3.3
+
+
+.. data:: WUNTRACED
+
+   This option for :func:`waitpid`, :func:`wait3`, and :func:`wait4` causes
+   child processes to also be reported if they have been stopped but their
+   current state has not been reported since they were stopped.
+
+   This option is not available for :func:`waitid`.
+
+   .. availability:: Unix, not Emscripten, not WASI.
+
+
+.. data:: WNOHANG
+
+   The option for :func:`waitpid`, :func:`wait3`, :func:`wait4`, and
+   :func:`waitid` to return immediately if no child process status is available
+   immediately.
+
+   .. availability:: Unix, not Emscripten, not WASI.
+
+
+.. data:: WNOWAIT
+
+   The option for :func:`waitid` to leave the child in a waitable state, so that
+   a later wait call can be used again to retrieve the child status information.
```

Comment:
Fix grammar, and I was unclear on the intended meaning here with more explicit syntax—do you mean a later `os.waitid` call?

```suggestion
   a later :func:`!waitid` call can be used to retrieve the child status information again.
```

...a later call to any `wait*` function?

```suggestion
   a later :func:`!wait*` call can be used to retrieve the child status information again.
```

...or something else, e.g. a later call to the underlying OS API (which could be stated more clearly)?